Index: ajax_poll.module
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- ajax_poll.module	(date 1466673780000)
+++ ajax_poll.module	(revision )
@@ -31,7 +31,7 @@
  */
 function ajax_poll_form_alter(&$form, $form_state, $form_id) {
   if (in_array($form_id, array('poll_view_voting', 'poll_cancel_form'))) {
-    $form['#attached']['js'][] = 'misc/jquery.form.js';
+    $form['#attached']['library'][] = array('system', 'jquery.form');
     $form['#attached']['js'][] = drupal_get_path('module', 'ajax_poll') . '/ajax_poll.js';
 
     $node = isset($form['#node']) ? $form['#node'] : node_load($form['#nid']);
